### ISSUE TITLE
Itemization of automatic_reconstructor command?

### ISSUE THREAD

**Issue by martinakos** [colmap/colmap#683]
If I run this command:

```
colmap automatic_reconstructor \
   --dense 0 \
   --workspace_path $DATASET_PATH \
   --image_path $DATASET_PATH/images \
   --data_type individual \
   --quality extreme \
   --single_camera 1
```

on a image set, I get a reasonable reconstruction. However, if I try to itemize the commands in the automatic_reconstructor I don't get the same reconstruction on the same data. 

This is my itemization taking into account the quality setting extreme,  the data_type and single_camera settings, according to [option_manager.cc](https://github.com/colmap/colmap/blob/dev/src/util/option_manager.cc)

```
colmap feature_extractor \
   --database_path $DATASET_PATH/database.db \
   --image_path $DATASET_PATH/images \ 
   --ImageReader.single_camera 1 \
   --SiftExtraction.estimate_affine_shape 1 \
   --SiftExtraction.domain_size_pooling 1 \      

colmap exhaustive_matcher \
   --database_path $DATASET_PATH/database.db \
   --SiftMatching.guided_matching 1
   
mkdir $DATASET_PATH/sparse

colmap mapper \
   --database_path $DATASET_PATH/database.db \
   --image_path $DATASET_PATH/images \
   --output_path $DATASET_PATH/sparse \
   --Mapper.ba_local_max_num_iterations 40 \
   --Mapper.ba_local_max_refinements 3 \
   --Mapper.ba_global_max_num_iterations 100 \
   --Mapper.min_focal_length_ratio 0.1 \
   --Mapper.max_focal_length_ratio 10 \
   --Mapper.max_extra_param 1.79769e+308
```

If I run these commands I don't get the same reconstruction as with the automatic_reconstructor, in fact most of the time I get the error

> #0   quit on numeric errors

Is there any other option I should set to reproduce the automatic_reconstructor command?

**Comment by ahojnnes** [comment]
Set use_pba 0

**Comment by martinakos** [comment]
Thanks! I added the --Mapper.ba_global_use_pba 0 to the mapper and that seems to eliminate the error. However, I still get different reconstructions compared to running the automatic_reconstructor command. In particular the camera poses I get from automatic_reconstructor are what I expect to get, but with my itemized script I get wrong camera poses.

Do you have any suggestions on what other parameters I could set to match the reconstructions?

**Comment by ahojnnes** [comment]
What do you mean by wrong? In the absolute or relative sense?

**Comment by martinakos** [comment]
In the relative way. I know the cameras are arranged in a circle around the object I'm reconstructing. With automatic_reconstructor I can see the resulting cameras in a circle but with the itemized steps I used above they are not in a circle, they are significantly spread around the expected circle. So I guess some other parameter or step must be different.

Having said that, I'm not so worried about replicating the automatic_reconstructor behaviour anymore. That was an intermediate step I tried as my particular use wasn't working initially, but now it does. So what I was looking for is to reconstruct from some images with known intrinsics and don't refine them during reconstruction. Using the itemized steps above plus setting the extra parameters results in the cameras being arranged in the expected circle. I'll go with that for the moment.
